feat: Remove JWT_AUTH_REFRESH_COOKIE Setting depr190

Yagnesh1998 opened the pull request:
**Description:**
The setting JWT_AUTH_REFRESH_COOKIE is meaningless and unused and should be cleaned up to avoid confusion.
In the very early days of introducing MFEs, we thought we were going to need this cookie in addition to the JWT cookie. However, it turned out we didn't need it, but the setting stuck around the contagion of it (being in cookiecutter and other template libraries) has resulted in it uselessly being copied to many repos.

**Supporting information:**
as per the original ticket https://github.com/openedx/public-engineering/issues/190, this setting is removed.

Agrendalath committed:
feat: remove field data binding from the runtime

Agrendalath committed:
feat: remove `field-data` service from runtime initialization

Agrendalath committed:
refactor: reuse services and wrappers between XBlocks

Commit:
test: remove unused helper functions from test_views.py

Since they were not used anywhere, it is ideal to delete them.

It would be redundant to keep them.

navinkarkera committed:
feat: configure acceptable language codes for youtube transcripts

Agrendalath committed:
refactor: remove redundant masquerade handling from runtime initialization

This part was added in 3d7246ecc81e3cb9e040b41f49c17e1da7e00b0a.
Then, 0dd4978f37b8d6d581e687a07ee5025b6693727d refactored the approach, but
did not remove these lines.

alex-sheehan-edx committed:
chore: bumping enterprise package version to 3.67.7

feanil committed:
fix: Don't assume there is a context.

Don't assume that there will be an extra `context` kwarg when using the
bookmark serializer.  We use it this way in the current code but that's
specific to us and not comon to all serializers.  There are a lot of API
documentation tools that automate introspecting serializers but they
won't know  that they have to send in a `context` to the serializer.

To make this serializer behave more like other serializers without
changing the behavior, we just need to check that the `context` value is
defined before we dig into it.  In the case that there is no `context`
we just treat it the same as if there is no `request` in the `context`.

KristinAoki committed:
fix: missing and broken waffle flag links (#32587)

feanil committed:
docs: Generate rst from the existing swagger docs.

Use an existing 3rd party library to generate RST docs from the existing
swagger.yaml file.

feanil committed:
docs: Remove the `api` docs sphinx project.

* This project was not being published anywhere.
* The previous commit that adds openapi generation capabilities to the
  `docs/guides` replaces what this was doing.

feanil committed:
docs: Rename swagger.yaml to lms-openapi.yaml

* Swagger was renamed to OpenAPI at some point so use the new name for
  clarity.

* Prefix with `lms` to make it clear that these are the APIs from the
  LMS and may not all be available in the CMS.

connorhaugh committed:
feat: ignore v2 xblocks in independence contract (#32602)

* feat: ignore v2 xblocks in independance contract

* docs: add documentation as to failure point

timmc-edx committed:
feat: Switch Dockerfile from `npm install` to `npm ci`; some cleanup (#32590)

Otherwise we're not really respecting the package-lock file and won't get
repeatable results.

Also:

- Clean up old error handling for npm<3. Were on npm 8 now. Probably
  can get rid of this.
- Use the shorthand `npm ci` rather than `npm clean-install` just for
  consistency with code elsewhere.
- Update comments in tests to be explicit about use of ci rather than
  install

johnnagro committed:
fix: improve username generation (#32613)

bradenmacdonald committed:
refactor: Split base.in into kernel.in (required) and bundled.in (optional) (#32552)

Plus remove a few unused and indirect dependencies

Commit:
chore(i18n): update translations

awais-ansari committed:
feat: add pagination in course enrollment list API (#32546)

* feat: add pagination in course enrollment list API

* refactor: enrollment course list API

* refactor: follow best practice in course enrollment list API

muhammadadeeltajamul committed:
feat: added management command to delete expired notifications (#32556)

github-actions[bot] committed:
feat: Upgrade Python dependency django (#32643)

security patch

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

Co-authored-by: awais786 <awais786@users.noreply.github.com>

muhammadadeeltajamul committed:
feat: added support for translation in notifications (#32626)

JodyBaileyy committed:
feat: added endpoint for context needed for recommendations experiment (#32645)

* feat: added endpoint for context needed for recommendations experiment

* chore: Removed unnecessary decorator

Commit:
feat: Remove JWT_AUTH_REFRESH_COOKIE Setting https://github.com/openedx/public-engineering/issues/190

openedx-webhooks commented:
Thanks for the pull request, @Yagnesh1998!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.




<!-- comment:external_pr -->
<!-- jenkins ok to test -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

Yagnesh1998 commented:
@e0d I think all checks have passed.

e0d commented:
@robrap and @dianakhuang I believe you were involved in scoping the issue that inspired this work.  OSPR is very small.  Can one of you review?

jajjibhai008 committed:
chore: bumping enterprise package version to 3.68.1

jajjibhai008 committed:
feat: Upgrade Python dependency edx-enterprise

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

Commit:
fix: mathjax resize on sindow resize (#32606)

* fix: mathjax resize on sindow resize

* fix: add resize for mathjax in wiki and cms

jesperhodge committed:
Refactor  rename contentstore xblock services (#32581)

This PR addresses the renaming of the contentstore/xblock_services folder to contentstore/xblock_storage_handlers as a follow-up to PR #32282. The renaming is done to prevent naming conflicts with xblock runtime services and to make the purpose of the files more understandable. The file xblock_service.py has been renamed to view_handlers.py to better reflect its functionality.

Justification and Future Refactoring Outlook:
The xblock_storage_handlers folder contains service methods that implement the business logic for view endpoints located in contentstore/views/block.py. It is renamed to xblock_storage_handlers to reflect its responsibility of handling storage-related operations of xblocks, such as creation, retrieval, and deletion.

The view_handlers.py file includes business methods called by the view endpoints. These methods, such as handle_xblock, delete_orphans, etc., interact with the required modulestore methods, handle any errors, and aggregate and serialize data for the response.

The term 'handler' in the context of 'view_handlers.py' represents methods that facilitate business logic for view endpoints. It is critical to note the distinction between these 'handler methods' and the xblock_handler method. The xblock_handler is a view endpoint itself, residing in contentstore/views/block.py, and is well known in this context. Although its name might suggest otherwise, it is not a handler in the sense of the 'handler methods' we've defined in 'view_handlers.py'. To maintain consistency with existing naming conventions, it remains as xblock_handler.

robrap commented:
@Yagnesh1998: Thank you. Would you mind adding the full link to the DEPR in your commit comment? Providing the ticket link in your commit description (not the title) is probably a good pattern to follow as you approach this work. Thanks again.

KristinAoki committed:
feat: add page redirect when mfe flag enabled (#32637)

Co-authored-by: Jesper Hodge <19345795+jesperhodge@users.noreply.github.com>

dianakhuang committed:
feat: Unpin SAML for upgrade. (#32443)

https://github.com/openedx/edx-platform/issues/32327

Commit:
chore: geoip2: update maxmind geolite country database

Yagnesh1998 committed:
Merge branch 'master' into edx-depr190

Yagnesh1998 commented:
I think there is some mistake I make another DEPR request

openedx-webhooks commented:
@Yagnesh1998 Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-platform&entry.867055334=https://github.com/openedx/edx-platform/pull/32647&entry.1484655318=https://github.com/Yagnesh1998&entry.752974735=2023-07-05+06:14&entry.1917517419=2023-07-06+05:37&entry.2133058324=No) so we can improve your experience in the future.

<!-- comment:end_survey -->
<!-- data: eyJqaXJhLXByZS1jbG9zZSI6IG51bGx9 -->
